CMakeLists.txt

```diff
@@ -212,13 +210,53 @@ include_directories( ${DBUS_CXX_INCLUDE_DIRECTORIES}
     ${sigc_INCLUDE_DIRS} )
 
 
+#
+# Dependencies
+#
+set( THREADS_PREFER_PTHREAD_FLAG ON )
+find_package( Threads REQUIRED )
+set( DBUS_CXX_LIBRARIES ${sigc_LIBRARIES} Threads::Threads )
+
+find_library( LIBRT rt )
+if( LIBRT )
+	list( APPEND DBUS_CXX_LIBRARIES ${LIBRT} )
+endif( LIBRT )
+
+if( WIN32 )
+    if( CMAKE_SYSTEM_VERSION )
+        set( ver ${CMAKE_SYSTEM_VERSION} )
+        string( REGEX MATCH "^([0-9]+).([0-9])" ver ${ver} )
+        string( REGEX MATCH "^([0-9]+)" verMajor ${ver} )
+        # Check for Windows 10, b/c we'll need to convert to hex 'A'.
+        if( "${verMajor}" MATCHES "10" )
+            set( verMajor "A" )
+            string( REGEX REPLACE "^([0-9]+)" ${verMajor} ver ${ver} )
+        endif()
+        # Remove all remaining '.' characters.
+        string( REPLACE "." "" ver ${ver} )
+        # Prepend each digit with a zero.
+        string( REGEX REPLACE "([0-9A-Z])" "0\\1" ver ${ver} )
+        set( WIN32_VER "0x${ver}" )
```

Comment:
Can we just set this to 0x0601(for Windows 7), or is there a specific reason this needs to be done?

Reply:
Unfortunately no, as Windows 10 is a special snowflake, hence the above voodoo.